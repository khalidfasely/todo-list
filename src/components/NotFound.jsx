import { Link } from "react-router-dom";

const NotFound = () => {
    return (
        <div>404 - Not Found <Link to='/home'>Go home.</Link></div>
    );
};

export default NotFound;