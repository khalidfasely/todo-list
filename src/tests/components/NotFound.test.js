import React from 'react';
import { render, screen } from '@testing-library/react';
import NotFound from '../../components/NotFound';

test('Should render NotFound component correctly', () => {
    const { asFragment } = render(<NotFound />);
    expect(asFragment()).toMatchSnapshot();
});

test('Should render go home link', () => {
    render(<NotFound />);
    const linkElement = screen.getByText(/go home./i);
    expect(linkElement).toBeInTheDocument();
});