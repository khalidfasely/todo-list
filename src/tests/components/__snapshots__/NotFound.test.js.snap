// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render NotFound component correctly 1`] = `
<DocumentFragment>
  <div>
    404 - Not Found 
    <a>
      Go home.
    </a>
  </div>
</DocumentFragment>
`;
